control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -1571,7 +1595,7 @@ func (r *reconciler) destroyCloudResources(ctx context.Context, hcp *hyperv1.Hos
 		// Here LastTransitionTime is used to indicate the last time the condition was updated by this
 		// controller. This is used by the CPO as a heartbeat to determine whether resource deletion is
 		// still in progress.
-		LastTransitionTime: metav1.Now(),
+		LastTransitionTime: ltt,
```

Comment:
why is this change needed?

Reply:
because we only updates the `LastTransitionTime` when "makes sense". if the conditions are not what we want, the controller will keep the one.